template/scripts/drupal-vm/config.yml

```diff
@@ -0,0 +1,26 @@
+# Update the hostname to the local development environment hostname
+vagrant_hostname: ${project.machine_name}.localhost
+vagrant_machine_name: ${project.machine_name}
+
+# Set drupal_site_name to the project's human-readable name.
+drupal_site_name: "${project.human_name}"
+
+# Provide the path to the project root to Vagrant.
+vagrant_synced_folders:
+  # Set the local_path for the first synced folder to `../`.
+  - local_path: ../
+    # Set the destination to the Acquia Cloud subscription machine name.
+    destination: ${project.machine_name}
+    type: nfs
+
+# Set drupal_core_path to the `destination` in the synced folder
+# configuration above, plus `/docroot`.
+drupal_core_path: "/var/www/${project.machine_name}/docroot"
+
+# If you build the makefile using BLT's built-in Phing task (recommended),
+# set `build_makefile` to `false`.
+build_makefile: false
```

Comment:
As of 3.1.0: I guess Drupal VM doesn't need to run `composer install` in this case so these two should be:

```yaml
drupal_core_path: "/var/www/${project.machine_name}/docroot"
build_composer_project: false
```

If you wanted to be more explicit the full working configuration with `composer install` running during `vagrant up`would be:

```yaml
build_composer_project: false
# toggling this to `true` would invoke `composer install` with the
# projects own `composer.json` successfully.
build_composer: false
drupal_composer_path: false
drupal_composer_install_dir: "/var/www/${project.machine_name}"
drupal_core_path: "{{ drupal_composer_install_dir }}/docroot"
```

Docs: http://docs.drupalvm.com/en/latest/deployment/composer/#using-composer-when-drupal-vm-is-a-composer-dependency-itself